src/coreclr/src/System.Private.CoreLib/src/System/GC.cs

```diff
@@ -54,28 +54,20 @@ public enum GCNotificationStatus
     public static class GC
     {
         [MethodImpl(MethodImplOptions.InternalCall)]
-        internal static extern void GetMemoryInfo(out ulong highMemLoadThresholdBytes,
-                                                  out ulong totalAvailableMemoryBytes,
-                                                  out ulong lastRecordedMemLoadBytes,
-                                                  out uint lastRecordedMemLoadPct,
-                                                  // The next two are size_t
-                                                  out UIntPtr lastRecordedHeapSizeBytes,
-                                                  out UIntPtr lastRecordedFragmentationBytes);
+        private static extern void GetMemoryInfo(GCMemoryInfoData data, int kind);
 
         public static GCMemoryInfo GetGCMemoryInfo()
         {
-            GetMemoryInfo(out ulong highMemLoadThresholdBytes,
-                          out ulong totalAvailableMemoryBytes,
-                          out ulong lastRecordedMemLoadBytes,
-                          out uint _,
-                          out UIntPtr lastRecordedHeapSizeBytes,
-                          out UIntPtr lastRecordedFragmentationBytes);
+            var data = new GCMemoryInfoData();
+            GetMemoryInfo(data, -1);
+            return new GCMemoryInfo(data);
+        }
 
-            return new GCMemoryInfo(highMemoryLoadThresholdBytes: (long)highMemLoadThresholdBytes,
-                                    memoryLoadBytes: (long)lastRecordedMemLoadBytes,
-                                    totalAvailableMemoryBytes: (long)totalAvailableMemoryBytes,
-                                    heapSizeBytes: (long)(ulong)lastRecordedHeapSizeBytes,
-                                    fragmentedBytes: (long)(ulong)lastRecordedFragmentationBytes);
+        public static GCMemoryInfo GetGCMemoryInfo(GCKind kind)
```

Comment:
Should this do any validation on GCKind to make sure it's one of the expected values?

Reply:
I do have an assert in the GC code but I agree, there should be one on the BCL side.